### PR TITLE
(require|disallow)SpacesInsideArrayBrackets: only check for ArrayExpressions, add new rules for Brackets

### PR DESCRIPTION
**Pull request by mikesherov**

### CODE REVIEW

**Comment by mikesherov** [comment]
ping @markelog ... should I also switch all of the presets to use the new rules? None of them have caveats for expression vs. accessor AFAIK.

File: lib/rules/disallow-spaces-inside-brackets.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+ * }
+ * ```
+ *
+ * ##### Valid for mode `true`
+ *
+ * ```js
+ * var x = [[1]];
+ * ```
+ *
+ * ##### Valid for mode `"allExcept"`
```
Comment (mrjoelkemp) [inline lib/rules/disallow-spaces-inside-brackets.js]: Can you add `"allExcept": [ "[", "]", "{", "}" ]` instead for clarity?

**Comment by mrjoelkemp** [comment]
:+1: Nice work!

File: lib/rules/disallow-spaces-inside-brackets.js (hunk 1 above)
Comment (mikesherov) [inline lib/rules/disallow-spaces-inside-brackets.js]: will do.

**Comment by markelog** [comment]
> ping @markelog ... should I also switch all of the presets to use the new rules? None of them have caveats for expression vs. accessor AFAIK.

Yeah.

I think after https://github.com/jscs-dev/node-jscs/pull/958#discussion_r23536197 we are good to go.

Nicely done!

**Comment by mikesherov** [comment]
@markelog should we be switching any presets over to the new rule? Or should I make a ticket for 2.0 to switch them over?

**Comment by markelog** [comment]
It seems new rules behave according to code styles.

But i also think we also should address https://github.com/jscs-dev/node-jscs/pull/822 and https://github.com/jquery/contribute.jquery.org/pull/104.

And do this sort of thing until 2.0 is released. After that, switch to strict release tactics.

**Comment by mikesherov** [comment]
@markelog OK, so I'm going to merge as is, and then file issue to add new rules later.

**Comment by zxqfox** [comment]
I'm sorry, but what about block comments between bracket and the next token?
```
var x = [/*1*/ 'a' ];
var x = [ /*2*/'a' ];
```
etc.

**Comment by mikesherov** [comment]
@zxqfox we can address that in another PR. This was about restoring old functionality.

**Comment by zxqfox** [comment]
almost managed ;-(

**Comment by mikesherov** [comment]
@zxqfox sorry, I pushed right before your comment! Block comments inside brackets is extremely rare... but you should file bugs against the existing rules.

**Comment by markelog** [comment]
> but you should file bugs against the existing rules.

+1

**Pull request closed without merging**